app/main.py

```diff
@@ -2,0 +1,37 @@
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if herbivore.hidden is False and isinstance(herbivore, Herbivore):
+            herbivore.health = herbivore.health - 50
+
+        if herbivore.health <= 0:
+            Animal.alive.remove(herbivore)
```

Comment:
Before removing the herbivore from `Animal.alive`, ensure that it is actually in the list. This will prevent potential errors if the `bite` method is called on an already dead herbivore.